src/libsyntax/attr/builtin.rs

```diff
@@ -461,8 +490,13 @@ pub fn cfg_matches(cfg: &ast::MetaItem, sess: &ParseSess, features: Option<&Feat
             MetaItemKind::List(..) => {
                 error(cfg.span, "unexpected parentheses after `cfg` predicate key")
             }
-            MetaItemKind::NameValue(lit) if !lit.node.is_str() => {
-                error(lit.span, "literal in `cfg` predicate value must be a string")
```

Comment:
Could you keep the old wording?
`AttrError::UnsupportedLiteral` can be extended to keep string fragments, and the resulting error message can be combined from them:
```
AttrError::UnsupportedLiteral(where, expected)
span_err("literal in {} must be a {}")

AttrError::UnsupportedLiteral("`cfg` predicate value", "string")
```

Reply:
It would give other `UnsupportedLiteral` errors better messages as well.